### DISCUSSION TITLE
How to set column names for pandas DataFrame?

### DISCUSSION THREAD

**Discussion by nandgator** [pydantic/pydantic#4737] [Question] [answered]
Example with a `dataclass`:

```python
from dataclasses import dataclass

from pandas import DataFrame

@dataclass
class Point:
    x: int
    y: int


print(DataFrame([Point(x=0, y=0), Point(x=2, y=1), Point(x=4, y=5)]))
```

Output

```
   x  y
0  0  0
1  2  1
2  4  5
```

---

Example with a `BaseModel`:

```python
from pydantic import BaseModel
from pandas import DataFrame

class Point(BaseModel):
    x: int
    y: int


print(DataFrame([Point(x=0, y=0), Point(x=2, y=1), Point(x=4, y=5)]))
```

Output

```
        0       1
0  (x, 0)  (y, 0)
1  (x, 2)  (y, 1)
2  (x, 4)  (y, 5)
```

---

**How do I get the same result as the one with `dataclass`?**

**Comment by samuelcolvin** [comment, marked as answer]
Simplest solution would be to use [pydantic's dataclasses](https://pydantic-docs.helpmanual.io/usage/dataclasses/), I've no idea how pandas detects and extracts attributes from dataclasses.

If there's some dunder method or similar we can add to support conversion to pandas dataframe rows, I'd be happy to add it - if you can find details, feel free to create a feature request issue.

**Reply by PrettyWood** [reply]
pandas supports dataclasses natively by running check `is_dataclass` and parsing `__dataclass_fields__`.
Best is to use pydantic dataclasses as @samuelcolvin said
